### PR TITLE
github actions and semantic-release

### PR DESCRIPTION
**Pull request by Enjoy2Live**

### CODE REVIEW

**Comment by chungthuang** [comment]
You also need https://github.com/dev-launchers/strapi/blob/main/package.json#L44-L59 and https://github.com/dev-launchers/strapi/blob/main/package.json#L19

File: .github/workflows/docker-build-release.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # GITHUB_REF is tag of the release https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release
+          ref: ${{ github.ref }}
+          # Number of commits to fetch. 0 indicates all history for all branches and tags.
+          fetch-depth: 0
+      - name: Docker meta
```
Comment (chungthuang) [inline .github/workflows/docker-build-release.yaml:16]: Let's use https://github.com/dev-launchers/strapi/blob/main/.github/workflows/production-image.yaml#L18-L24 for tagging.

File: .github/workflows/docker-build-release.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
```
Comment (chungthuang) [inline .github/workflows/docker-build-release.yaml:36]: Dockerfile is not defined yet.

File: .github/workflows/docker-build-release.yaml (hunk 1 above)
Comment (Enjoy2Live) [reply .github/workflows/docker-build-release.yaml:16]: I think I've done what you requested with this [commit](https://github.com/dev-launchers/site-projects/pull/16/commits/d5161093eb91d72aaf92c5cb4bc8f75b1a4efc13)

File: .github/workflows/docker-build-release.yaml (hunk 2 above)
Comment (Enjoy2Live) [reply .github/workflows/docker-build-release.yaml:36]: @chungthuang the dockerfile I have in the last [commit](https://github.com/dev-launchers/site-projects/pull/16/commits/d5161093eb91d72aaf92c5cb4bc8f75b1a4efc13) handles environment variables for production. I don't know if it does it correctly but I took it from next.js official docs.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+on: [pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
```
Comment (chungthuang) [inline .github/workflows/test.yaml:11]: Your Dockerfile is using node 14, do you want to upgrade the version here?

**Review by chungthuang — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 3 above)
Comment (Enjoy2Live) [reply .github/workflows/test.yaml:11]: yeah I'll change it here too, thanks for noticing that.

**Pull request merged**